### PR TITLE
Feature/split haproxy processes

### LINKED ISSUE #3012: MultiTenant: Inception while starting haproxy in cluster
While starting haproxy, it tries to connect to haproxy to get the tenants and their potential domain name.

In a cluster, the DB is only up after haproxy starts.

We should either remove this feature, or make haproxy DB and portal processes completely separate

Kind of like ping pong where there are 2 different parties

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Separate the haproxy process of the DB and the one of the portal since the portal frontend now requires access to the DB (see #3012).

# Impacts
Haproxy

# Issue
fixes #3012 

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* The haproxy processes (DB access and portal access) have been separated into 2 different services (haproxy-portal and haproxy-db)

### CODE REVIEW

**Comment by julsemaan** [comment]
@jrouzierinverse 

Could you take a look at this PR

File: lib/pf/services/manager/haproxy_db.pm [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    $self->executable . " -f $generated_conf_dir/haproxy-db.conf -p $install_dir/var/run/haproxy-db.pid";
+}
+
+has '+shouldCheckup' => ( default => sub { 0 }  );
+
+sub executable {
+    my ($self) = @_;
+    my $service = ( $Config{'services'}{"haproxy-db_binary"} || "$install_dir/sbin/haproxy" );
+    return $service;
+}
```
Comment (jrouzierinverse) [inline lib/pf/services/manager/haproxy_db.pm]: Put this in the base class.

```perl
sub _build_executable {
    my ($self) = @_;
    my $name = $self->name;
    my $service = ( $Config{'services'}{"${name}_binary"} || "$install_dir/sbin/haproxy" );
    return $service;
}
```

File: lib/pf/services/manager/haproxy_portal.pm [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    $self->executable . " -f $generated_conf_dir/haproxy-portal.conf -p $install_dir/var/run/haproxy-portal.pid";
+}
+
+has '+shouldCheckup' => ( default => sub { 0 }  );
+
+sub executable {
+    my ($self) = @_;
+    my $service = ( $Config{'services'}{"haproxy-portal_binary"} || "$install_dir/sbin/haproxy" );
+    return $service;
+}
```
Comment (jrouzierinverse) [inline lib/pf/services/manager/haproxy_portal.pm]: Remove this and use the base class

File: lib/pf/services/manager/haproxy_db.pm [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+extends 'pf::services::manager::haproxy';
+
+has '+name' => (default => sub { 'haproxy-db' } );
+
+has '+haproxy_config_template' => (default => sub { "$conf_dir/haproxy-db.conf" });
+
+sub _cmdLine {
+    my $self = shift;
+    $self->executable . " -f $generated_conf_dir/haproxy-db.conf -p $install_dir/var/run/haproxy-db.pid";
+}
```
Comment (jrouzierinverse) [inline lib/pf/services/manager/haproxy_db.pm]: Refactor this to the base class
```perl
sub _cmdLine {
    my $self = shift;
    my $name = $self->name;
    $self->executable . " -f $generated_conf_dir/$name.conf -p $install_dir/var/run/$name.pid";
}

```

File: lib/pf/services/manager/haproxy_portal.pm [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+);
+
+has '+name' => (default => sub { 'haproxy-portal' } );
+
+has '+haproxy_config_template' => (default => sub { "$conf_dir/haproxy-portal.conf" });
+
+sub _cmdLine {
+    my $self = shift;
+    $self->executable . " -f $generated_conf_dir/haproxy-portal.conf -p $install_dir/var/run/haproxy-portal.pid";
+}
```
Comment (jrouzierinverse) [inline lib/pf/services/manager/haproxy_portal.pm]: Use the base class version

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]
Looks good just requires little refactoring.

**Comment by julsemaan** [comment]
@jrouzierinverse 

Changes done

**Pull request merged**